t/re-split.t

```diff
@@ -0,0 +1,707 @@
+                return
+            end
+
+            for i = 1, #my_table do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+,
+b
+,
+c,d,e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: split matches, ctx arg
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d,e", ",", nil, { pos = 5 })
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+c
+d
+e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: split matches, trailing subjects
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split(",a,b,c,d,", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                if res[i] == "" then
+                    ngx.say("_blank_")
+                else
+                    ngx.say(res[i])
+                end
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+_blank_
+a
+b
+c
+d
+_blank_
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+attempt to get length of local 'regex' (a number value)
+
+
+
+=== TEST 18: split matches, real use-case
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd,erfg,ghij;hello world;aaa", ",|;")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+erfg
+ghij
+hello world
+aaa
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 19: split no matches
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 20: subject is not a string type
+--- http_config eval: $::HttpConfig
```

Comment:
Recompiled with no pool:
```
$ PATH="/usr/local/openresty-valgrind/nginx/sbin:$PATH" nginx -V                                    
nginx version: openresty/1.11.2.1 (no pool)
built by gcc 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3) 
built with OpenSSL 1.0.2j  26 Sep 2016
TLS SNI support enabled
configure arguments: --prefix=/usr/local/openresty-valgrind/nginx --with-debug --with-cc-opt='-DNGX_LUA_USE_ASSERT -DNGX_LUA_ABORT_AT_PANIC -O2' --add-module=../ngx_devel_kit-0.3.0 --add-module=../echo-nginx-module-0.60 --add-module=../xss-nginx-module-0.05 --add-module=../ngx_coolkit-0.2rc3 --add-module=../set-misc-nginx-module-0.31 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.06 --add-module=../srcache-nginx-module-0.31 --add-module=../ngx_lua-0.10.6 --add-module=../ngx_lua_upstream-0.06 --add-module=../headers-more-nginx-module-0.31 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.17 --add-module=../redis2-nginx-module-0.13 --add-module=../redis-nginx-module-0.3.7 --add-module=../rds-json-nginx-module-0.14 --add-module=../rds-csv-nginx-module-0.07 --with-ld-opt=-Wl,-rpath,/usr/local/openresty-valgrind/luajit/lib --with-openssl=/home/thibaultcha/tmp/openssl-1.0.2j --with-http_ssl_module
```

But Valgrind still does not complain, sorry I can't seem to be of help here.